stripe/src/main/java/com/stripe/android/model/SourceCardData.java

```diff
@@ -0,0 +1,306 @@
+        StripeNetworkUtils.removeNullParams(objectMap);
+        return objectMap;
+    }
+
+    private SourceCardData setAddressLine1Check(String addressLine1Check) {
+        mAddressLine1Check = addressLine1Check;
+        return this;
+    }
+
+    private SourceCardData setAddressZipCheck(String addressZipCheck) {
+        mAddressZipCheck = addressZipCheck;
+        return this;
+    }
+
+    private SourceCardData setBrand(String brand) {
+        mBrand = brand;
+        return this;
+    }
+
+    private SourceCardData setCountry(String country) {
+        mCountry = country;
+        return this;
+    }
+
+    private SourceCardData setCvcCheck(String cvcCheck) {
+        mCvcCheck = cvcCheck;
+        return this;
+    }
+
+    private SourceCardData setDynamicLast4(String dynamicLast4) {
+        mDynamicLast4 = dynamicLast4;
+        return this;
+    }
+
+    private SourceCardData setExpiryMonth(Integer expiryMonth) {
+        mExpiryMonth = expiryMonth;
+        return this;
+    }
+
+    private SourceCardData setExpiryYear(Integer expiryYear) {
+        mExpiryYear = expiryYear;
+        return this;
+    }
+
+    private SourceCardData setFunding(String funding) {
+        mFunding = funding;
+        return this;
+    }
+
+    private SourceCardData setLast4(String last4) {
+        mLast4 = last4;
+        return this;
+    }
+
+    private SourceCardData setThreeDSecureStatus(String threeDSecureStatus) {
+        mThreeDSecureStatus = threeDSecureStatus;
+        return this;
+    }
+
+    private SourceCardData setTokenizationMethod(String tokenizationMethod) {
+        mTokenizationMethod = tokenizationMethod;
+        return this;
+    }
+
+    @Nullable
+    static SourceCardData fromJson(@Nullable JSONObject jsonObject) {
+        if (jsonObject == null) {
+            return null;
+        }
+
+        SourceCardData cardData = new SourceCardData();
+        cardData.setAddressLine1Check(optString(jsonObject, FIELD_ADDRESS_LINE1_CHECK))
+                .setAddressZipCheck(optString(jsonObject, FIELD_ADDRESS_ZIP_CHECK))
+                .setBrand(StripeTextUtils.asCardBrand(optString(jsonObject, FIELD_BRAND)))
+                .setCountry(optString(jsonObject, FIELD_COUNTRY))
+                .setCvcCheck(optString(jsonObject, FIELD_CVC_CHECK))
+                .setDynamicLast4(optString(jsonObject, FIELD_DYNAMIC_LAST4))
+                .setExpiryMonth(optInteger(jsonObject, FIELD_EXP_MONTH))
+                .setExpiryYear(optInteger(jsonObject, FIELD_EXP_YEAR))
+                .setFunding(StripeTextUtils.asFundingType(optString(jsonObject, FIELD_FUNDING)))
+                .setLast4(optString(jsonObject, FIELD_LAST4))
+                .setThreeDSecureStatus(asThreeDSecureStatus(optString(jsonObject,
+                        FIELD_THREE_D_SECURE)))
+                .setTokenizationMethod(optString(jsonObject, FIELD_TOKENIZATION_METHOD));
+
+        Map<String, Object> nonStandardFields =
```

Comment:
Actually, no -- I hadn't thought to do it with the ones that had well-documented specs.